src/arena.c

```diff
@@ -697,7 +767,13 @@ arena_reset(tsd_t *tsd, arena_t *arena) {
 		if (config_prof && opt_prof) {
 			prof_free(tsd, ptr, usize, &alloc_ctx);
 		}
-		large_dalloc(tsd_tsdn(tsd), edata);
+		if (config_prof && opt_prof && alloc_ctx.szind < nhbins &&
```

Comment:
The `alloc_ctx.szind < nhbins` check isn't necessary? We pass `tcache = NULL` to arena_dalloc_promoted_impl so it can't go through the thread cache path.